### ISSUE TITLE
new release?

### ISSUE THREAD

**Issue by jvolkening** [finance-quote/finance-quote#154]
Are there any plans to push a new release to CPAN soon, or any reason not to? I'm dealing with issue #148 which has been fixed but not made it into CPAN yet. In the meantime I can install from here.

**Comment by jvolkening** [comment]
Actually, I found when installing from the git master branch that many tests are failing on my system. Specifically 36/67 test units fail:

```
t/asegr.t	fail
t/aufunds.t	fail
t/bmonesbittburns.t	fail
t/bsero.t	fail
t/cdnfundlibrary.t	fail
t/citywire.t	fail
t/cominvest.t	fail
t/cse.t	fail
t/currencies.t	fail
t/currency.t	fail
t/dws.t	fail
t/fool.t	fail
t/ftfunds.t	fail
t/indiamutual.t	fail
t/lerevenu.t	fail
t/morningstarau.t	fail
t/morningstarCH.t	fail
t/morningstar.t	fail
t/mstaruk.t	fail
t/platinum.t	fail
t/seb.t	fail
t/sixfunds.t	fail
t/sixshares.t	fail
t/stockhousecanada.t	fail
t/tdefunds.t	fail
t/tdwaterhouse.t	fail
t/tiaacref.t	fail
t/tnetuk.t	fail
t/troweprice.t	fail
t/trustnet.t	fail
t/tsx.t	fail
t/vanguard.t	fail
t/vwd.t	fail
t/yahoojson.t	fail
t/za.t	fail
t/za_unittrusts.t	fail
t/00-use.t	pass
t/01-pod.t	pass
t/02-pod-coverage.t	pass
t/03-kwalitee.t	pass
t/04-critic.t	pass
t/05-data-dumper.t	pass
t/aex.t	pass
t/aiahk.t	pass
t/alphavantage.t	pass
t/asx.t	pass
t/bourso.t	pass
t/currency_lookup.t	pass
t/deka.t	pass
t/fidelityfixed.t	pass
t/fidelity.t	pass
t/financecanada.t	pass
t/finanzpartner.t	pass
t/ftportfolios.t	pass
t/goldmoney.t	pass
t/hex.t	pass
t/hu.t	pass
t/iexcloud.t	pass
t/maninvestments.t	pass
t/morningstarJP.t	pass
t/nzx.t	pass
t/oslobors.t	pass
t/quote.t	pass
t/tsp.t	pass
t/ukfunds.t	pass
t/union.t	pass
t/usfedbonds.t	pass
```
I can see in the source for some of these that they require authentication tokens, so maybe cannot be tested without an account. For several others that I looked at, however, I found a combination of minor fixable bugs (e.g. YahooJSON), broken source URLs (e.g. T Rowe Price) or minor issues in the test suite (e.g. Motley Fool). 

I would be happy to put together PRs for some of these, since I use F::Q as part of GnuCash, but first I'd like to confirm that this is really the state of the code and not just an issue specific to my system. If so, is there a way already to disable tests for sources that require a paid account?

**Comment by vincentl** [comment]
Finance::Quote development has slow down quite a bit. I've been trying to work through open pull requests and outstanding issues before proposing a new release. It would be great if you have some time to also work on FQ.  Do you want to just open pull requests or do you want to be added to the project?

There are 2 modules that require API keys (AlphaVantage and IEXCloud) for online tests.  With those two keys set, I get 35 tests with 1 more failures.

There is a project for release 1.50 (https://github.com/finance-quote/finance-quote/projects/2).  Addressing open issues, reviewing pull requests, or investigating test failures are all helpful.

**Comment by jvolkening** [comment]
Okay, great -- I hadn't looked at the project page. I see most/all of the failures I see are already on there. I can just open PRs case-by-case, but you can add me to the project if you'd like.

**Comment by vincentl** [comment]
@ecocode or @pjf - can we add @jvolkening as a member?

**Comment by vincentl** [comment]
Thanks for the pull requests.  I merged the first 2.  

Please pull from master before making more changes because I just merged a pull request that implemented some major changes.

**Comment by jvolkening** [comment]
> Please pull from master before making more changes because I just merged a pull request that implemented some major changes.

Okay. I just opened several more PRs to fix minor issues causing test failures, and I had to merge upstream master back into them before opening the PR, but there were no conflicts.

I wonder...it seems that at least one (T Rowe Price) and possibly more of the data sources based on screen scraping are broken because of dead URLs. Is it really worth trying to update these, given that for some at least the same data is available via Yahoo JSON or possibly other, more stable, API-based sources?

**Comment by bpschuck** [comment]
Some months back; Erik, Vincent, and myself were discussing doing a cleanup where many of the non-working modules were going to be removed. Related to this is part of the reason the Modules-README.yml file was created. Assumption being that they have been broken so long and only a few had issues reported about them. Some of those issues were in fact closed by directing the user to Yahoo JSON as an alternative source of the data.

**Comment by vincentl** [comment]
I just added a task to the 1.50 project to validate Modules-README.yml.

For modules with no hope (see issue #152 for example), suggest that we 
  1. git rm the module & test file, update Quote.pm & commit
  2. Update Modules-README.yml by changing "state" to "removed" and set the "remove" key to the commit hash from step 1

At one point TIAA fund prices were only available from TIAA.  Not sure about T Rowe Price.  Maybe take some of the test symbols from the about to be removed module test file and try them with some of the more general modules to see if there is an alternative source.

**Comment by ecocode** [comment]
vincentl <notifications@github.com> writes:

> @ecocode or @pjf - can we add @jvolkening as a member?

Hello,

I sent an invitation to @jvolkening.

best
--
erik colson

**Comment by vincentl** [comment]
Thanks @ecocode for inviting @jvolkening. I'm happy to be listed as an owner for this project.

**Comment by vincentl** [comment]
@bpschuck - I just pushed an update to the pod documentation for Quote.pm for the new API changes.  The README.md automatically gets updated.  To you mind taking a look at the updated documentation?

Feel free to make changes in Quote.pm and push the changes, or we could open a new issue to discuss the documentation changes.  I writing the documentation, I realized that there are several functions that appeared to be "public," so I kept them that way, but I don't see any reason someone would call those functions.  Interested in your thoughts.

**Comment by vincentl** [comment]
typed the wrong issue number in my last commit...reopening

**Comment by vincentl** [comment]
Still working through the backlog of merge requests.  

I've proposed a strategy for how to include "development & debugging" code in module files in the Hackers-Guide.

After merging or closing the remaining merge requests, will turn to cleaning up the tests.

Also working on a script to help maintain Modules-README.yml.

**Comment by vincentl** [comment]
No problem.  Focus on you and get well!

**Comment by vincentl** [comment]
With merge 6e814cc0730eb0979d6dbb944fbf711869d1cf50 that introduced Finance::Quote::CurrencyRates and multiple currency rate backends, we are getting closer to 1.50.

Following the same scheme for how callers can indicate the failover order for CurrencyRate modules, I want to extend that to fetch methods too.  Once that is done, still need to clean up failing tests and tidy up Modules-README.yml.

**Comment by vincentl** [comment]
I think commit 15e5cfad047af6c359f82482b15e0c5d5e8663a5 is ready for others to check for issues before a 1.50 release!

**Comment by bpschuck** [comment]
After some offline discussions with Erik, we may be uploading a new release to CPAN soon.
Please stand by.

**Comment by vincentl** [comment]
@bpschuck - I updated [Project 1.50](https://github.com/finance-quote/finance-quote/projects/2) to reflect the remaining 7 items that should be addressed before releasing 1.50.  If you are able to release soon, I can try to clean these up over the next week.

**Comment by bpschuck** [comment]
Vincent,

Since almost all the previous releases were done with PRs and Issues 
outstanding, in my opinion I do not see the need to have almost all of 
them addressed. Especially for issues that have existed for years in the 
last half dozen or so releases.

Being more pragmatic, I tend to see the incorporation of many changes as 
increasing the probability of introducing new bugs.

- Bruce S.

**Comment by vincentl** [comment]
I handled the really easy ones.  Issue #174 will take more work, so we can push that to later.  Pull request #177 isn't complete and only affects online tests.

**Comment by bpschuck** [comment]
Awesome!
Perhaps you could update the `Changes` file, at least with any of the major updates you've made since v1.49 was released? Don't go nuts trying to recall everything in the past year plus. Thanks.
In hindsight, it looks like updating this as issues and PRs are closed would have been a good idea. Such is life.

**Comment by bpschuck** [comment]
@vincentl - I see you added a stanza for FTPortfolios.pm in Modules-README.yml. But there isn't a module with that name? Did you perhaps mean FTfunds.pm?

**Comment by vincentl** [comment]
@bpschuck - FTPortfolios as removed in commit 2f0f6bdd8664d15ece3ff11a16ad42a647b9fa3b because the site appears to require javascript on the client side to fetch quotes.

**Comment by bpschuck** [comment]
So Modules-README.yml is incorrect and needs to be updated.

- module: FTPortfolios.pm
   state: working
   added: TBD
   changed: ~
   removed: ~
   urls:
   apikey: false
   notes: ~
   testfile: TBD
   testcases:

**Comment by bpschuck** [comment]
@vincentl 
If it's okay with you, I am going to close this issue now that v1.50 has been uploaded to CPAN.

**Comment by vincentl** [comment]
@bpschuck - agree and thanks!
